frappe/core/doctype/doctype/doctype.py

```diff
@@ -185,6 +185,11 @@ def after_insert(self):
 		# clear user cache so that on the next reload this doctype is included in boot
 		clear_user_cache(frappe.session.user)
 
+	def after_save(self):
+		if not self.is_new():
+			if self.has_value_changed("istable"):
+				clear_user_cache(frappe.session.user)
```

Comment:
Why not do this always after save? 

```suggestion
			clear_user_cache(frappe.session.user)
```

Reply:
also `on_update` instead of `after_save`

Reply:
removed `after_insert` and just did that in `on_update` hook